addons/point_of_sale/static/src/app/barcode_reader_service.js

```diff
@@ -0,0 +1,16 @@
+/** @odoo-module */
+
+import { registry } from "@web/core/registry";
+import { BarcodeReader } from "@point_of_sale/js/barcode_reader";
+import BarcodeParser from "barcodes.BarcodeParser";
+
+export const barcodeReader = {
+    async start(env) {
+        // TODO-JCB: Should error if there is no nomenclature_id.
```

Comment:
I tried simple throwing of error here but it doesn't seem to work. The app continues to work and the barcode_reader is not loaded. How do you think should I handle the error here?

Reply:
Hmm, maybe we can just prevent the loading of pos if during the rendering of "point_of_sale.index", the `nomenclature_id` is not present.